From 1306135e4898b78f8f371d072b622758a6a0fe55 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 24 Apr 2023 15:37:18 +0800
Subject: [PATCH] pfeng: fix build errors to compatible with v6.1 kernel

Issue: LINUXEXEC-27398

Now the linux kernel has been updated to v6.1, so update pfeng driver codes
to compatible with it.

Or else, there will be build errors when compiling the pfeng recipes.
This patch will fix the build issues.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 sw/linux-pfeng/pfeng-hif.c      | 2 +-
 sw/linux-pfeng/pfeng-netif.c    | 2 +-
 sw/oal/public/oal_types_linux.h | 2 +-
 sw/pfe_platform/src/pfe_hm.c    | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sw/linux-pfeng/pfeng-hif.c b/sw/linux-pfeng/pfeng-hif.c
index b815468..c3248a9 100644
--- a/sw/linux-pfeng/pfeng-hif.c
+++ b/sw/linux-pfeng/pfeng-hif.c
@@ -597,7 +597,7 @@ static int pfeng_hif_chnl_drv_create(struct pfeng_priv *priv, u32 idx)
 	pfeng_hif_chnl_set_coalesce(chnl, priv->clk_sys, PFENG_INT_TIMER_DEFAULT, 0);
 
 	chnl->status = PFENG_HIF_STATUS_ENABLED;
-	netif_napi_add(&chnl->dummy_netdev, &chnl->napi, pfeng_hif_chnl_poll, NAPI_POLL_WEIGHT);
+	netif_napi_add(&chnl->dummy_netdev, &chnl->napi, pfeng_hif_chnl_poll);
 	napi_enable(&chnl->napi);
 
 	HM_MSG_DEV_INFO(dev, "HIF%d enabled\n", idx);
diff --git a/sw/linux-pfeng/pfeng-netif.c b/sw/linux-pfeng/pfeng-netif.c
index 156ebe0..79829c8 100644
--- a/sw/linux-pfeng/pfeng-netif.c
+++ b/sw/linux-pfeng/pfeng-netif.c
@@ -806,7 +806,7 @@ static int pfeng_netif_set_mac_address(struct net_device *netdev, void *p)
 	struct sockaddr *addr = (struct sockaddr *)p;
 
 	if (is_valid_ether_addr(addr->sa_data)) {
-		ether_addr_copy(netdev->dev_addr, addr->sa_data);
+		eth_hw_addr_set(netdev, addr->sa_data);
 	} else {
 		HM_MSG_NETDEV_WARN(netdev, "No MAC address found, using random\n");
 		eth_hw_addr_random(netdev);
diff --git a/sw/oal/public/oal_types_linux.h b/sw/oal/public/oal_types_linux.h
index 0022de1..3f822eb 100644
--- a/sw/oal/public/oal_types_linux.h
+++ b/sw/oal/public/oal_types_linux.h
@@ -10,7 +10,7 @@
 
 #include <linux/kernel.h>
 #include <linux/types.h>
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #include <linux/string.h>
 #include <linux/platform_device.h>
diff --git a/sw/pfe_platform/src/pfe_hm.c b/sw/pfe_platform/src/pfe_hm.c
index 2928f03..f23f479 100644
--- a/sw/pfe_platform/src/pfe_hm.c
+++ b/sw/pfe_platform/src/pfe_hm.c
@@ -12,7 +12,7 @@
 #include "pfe_hm.h"
 
 #ifndef PFE_CFG_TARGET_OS_AUTOSAR
-#include <stdarg.h>
+#include <linux/stdarg.h>
 #endif
 
 #define ARRAY_LEN(x) (sizeof(x)/sizeof(x[0]))
-- 
2.25.1

